packages/terra-wallet-provider/lib/TerraWalletProvider.ts

```diff
@@ -81,12 +84,13 @@ export default class TerraWalletProvider extends WalletProvider {
     }
 
     const wallet = new MnemonicKey({
-      mnemonic: this._mnemonic
+      mnemonic: this._mnemonic,
+      index: this._index
     })
 
     const result = new Address({
       address: wallet.accAddress,
-      derivationPath: this._baseDerivationPath + `/0/0`,
+      derivationPath: this._baseDerivationPath + `/0/${this._index}`,
```

Comment:
In the other providers, we have requested for the full path to be passed, instead of separate base + index
I think this is best because it gives flexibility to the client

Reply:
Changed applied